pkg/kubenest/controller/virtualcluster_init_controller.go

```diff
@@ -104,7 +106,11 @@ func (c *VirtualClusterInitController) Reconcile(ctx context.Context, request re
 		if err != nil {
 			return reconcile.Result{RequeueAfter: RequeueTime}, errors.Wrapf(err, "Error update virtualcluster %s status", updatedCluster.Name)
 		}
-
+		//get latest virtualcluster
+		if err = c.Get(ctx, request.NamespacedName, originalCluster); err != nil {
+			return reconcile.Result{RequeueAfter: RequeueTime}, nil
+		}
```

Comment:
If virtualCluster is not found, it should be considered normal instead of requeue.